runtime/tools/python/ttrt/common/memory.py

```diff
@@ -0,0 +1,82 @@
+# SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import json
+import importlib.machinery
+import sys
+import signal
+import os
+import io
+import subprocess
+import time
+import socket
+from pkg_resources import get_distribution
+import shutil
+import atexit
+import re
+from functools import partial
+
+from ttrt.common.util import *
+
+
+class MemoryRuntimeConfig:
+    def __init__(self, device=None, artifact_dir="", tt_metal_home_path=""):
+        self.device = device
+        self.artifact_dir = artifact_dir
+        self.tt_metal_home_path = tt_metal_home_path
+
+
+def memory(memory_runtime_config, binary, program_context, op_context):
+    device = memory_runtime_config.device
+
+    print("-----------executing memory dump-----------")
+
+    try:
+        import ttrt.runtime
+        import ttrt.binary
+
+        device.dump_memory_report()
+
+        op_debug_str = ttrt.runtime.get_op_debug_str(op_context)
```

Comment:
Should we just use `get_op_loc_info` instead of trying to parse the loc out of the debug string?

Reply:
agree. fixed